development/nautobot_config.py

```diff
@@ -217,6 +217,7 @@ def is_truthy(arg):
         "enable_webex": True,
         "microsoft_app_id": os.environ.get("MICROSOFT_APP_ID"),
         "microsoft_app_password": os.environ.get("MICROSOFT_APP_PASSWORD"),
+        "microsoft_max_msg_size": int(os.environ.get("MICROSOFT_MAX_MSG_SIZE", 0)),
```

Comment:
Do we know what the actual max message size is for MS Teams? Does this even need to be a configurable parameter?

Reply:
From MS Teams: 28 KB is an approximate limit because it includes the message itself (text, image links, etc.), @-mentions, number of connectors, and reactions.

Looks like character length has nothing to do with this.  Maybe there is a message coming back to the chatops plugin from Microsoft that would be better used.

Reply:
I've not observed any messages coming back from Microsoft. I agree that counting character length is not ideal but in our case I believe it will the main contributor to the message size. From testing it looked to me that messages started getting lost when content length exceeded approx. 21000.

Reply:
@glennmatthews @smk4664 I did some more digging and I identified some responses we could use, but they're not consistent.

E.g. when using `nautobot get-device-facts` with large number of devices I get http code 413 back with a telling message:

```
worker_1    | 19:10:38 INFO Response code 413 - rq.worker - ms_teams - /source/nautobot_chatops/dispatchers/ms_teams.py:111
worker_1    | b'{"error":{"code":"MessageSizeTooBig","message":"Message size too large."}}'
```

When I ran `nautobot get-vlans site <site-name>` for site which had a lot of vlans a message header made it to the chat but VLAN list did not and I could see in debug http code 400 with a generic error message:

```
worker_1    | 19:10:06 INFO Response code 400 - rq.worker - ms_teams - /source/nautobot_chatops/dispatchers/ms_teams.py:111
worker_1    | b'{"error":{"code":"ServiceError","message":"Unknown"}}'
```